conjureup/utils.py

```diff
@@ -319,7 +348,19 @@ def setup_metadata_controller():
         bundle_filename = charm.get_bundle(
             app.config['metadata']['bundle-location'], True)
 
-    bundle = Bundle(filename=bundle_filename)
+    # Load bundle data early so we can merge any additional charm options
+    with open(bundle_filename) as f:
+        bundle_data = yaml.load(f)
+
+    bundle_custom_filename = os.path.join(app.config['spell-dir'],
+                                          'bundle-custom.yaml')
+    if os.path.isfile(bundle_custom_filename):
+        with open(bundle_custom_filename) as f:
+            bundle_custom = yaml.load(f)
```

Comment:
I find [pathlib](https://docs.python.org/3/library/pathlib.html) to be a bit nicer, but that's just personal preference:

```python
bundle_custom_file = Path(app.config['spell-dir']) / 'bundle-custom.yaml'
if bundle_custom_file.is_file():
    bundle_custom = yaml.safe_load(bundle_custom_file.read_text())
    bundle_data = merge_dicts(bundle_data, bundle_custom)
```